app/controllers/podcast_metrics_controller.rb

```diff
@@ -195,46 +214,39 @@ def set_podcast
     render_not_found(e)
   end
 
-  def set_date_range
-    @date_start = metrics_params[:date_start]
-    @date_end = metrics_params[:date_end]
-    @interval = metrics_params[:interval]
-    @date_range = generate_date_range(@date_start, @date_end, @interval)
+  def metrics_params
+    params
+      .permit(:podcast_id, :episode_id, :prev_episode_id)
   end
 
-  def set_uniques
-    @uniques_selection = uniques_params[:uniques_selection]
-  end
+  def calculate_episode_trend(episode, prev_episode)
+    return nil unless episode.first_rss_published_at.present? && prev_episode.present?
+    return nil if (episode.first_rss_published_at + 1.day) > Time.now
 
-  def set_dropday_range
-    @dropday_range = dropdays_params[:dropday_range]
-  end
+    ep_dropday_sum = episode_dropday_query(episode)
+    previous_ep_dropday_sum = episode_dropday_query(prev_episode)
 
-  def metrics_params
-    params
-      .permit(:podcast_id, :date_start, :date_end, :interval)
-      .with_defaults(
-        date_start: 30.days.ago.utc_date,
-        date_end: Date.utc_today,
-        interval: "DAY"
-      )
+    return nil if ep_dropday_sum <= 0 || previous_ep_dropday_sum <= 0
+
+    ((ep_dropday_sum.to_f / previous_ep_dropday_sum.to_f) - 1).round(3)
   end
 
-  def uniques_params
-    params
-      .permit(:uniques_selection)
-      .with_defaults(
-        uniques_selection: "last_7_rolling"
-      )
-      .merge(metrics_params)
+  def episode_dropday_query(ep)
+    lowerbound = publish_hour(ep)
+    upperbound = lowerbound + 24.hours
+
+    Rollups::HourlyDownload
+      .where(episode_id: ep[:guid], hour: (lowerbound...upperbound))
+      .final
+      .load_async
+      .sum(:count)
   end
 
-  def dropdays_params
-    params
-      .permit(:dropday_range, :interval)
-      .with_defaults(
-        dropday_range: 7
-      )
-      .merge(metrics_params)
+  def publish_hour(episode)
```

Comment:
is this accurate? trying to address the `.beginning_of_hour` comment but also catching episodes that don't have `.first_rss_published_at`.

Reply:
I would probably just return a 0 / empty array, instead of querying clickhouse at all, if that's nil.

The implication is, this episode was never written to the default feed.  So you're not going to find any data in Clickhouse.